### DISCUSSION TITLE
Langium Long-Term Roadmap

### DISCUSSION THREAD

**Discussion by msujew** [eclipse-langium/langium#646] [Announcements]
For the latest milestone and what we have planned short-term for Langium, see [here](https://github.com/langium/langium/discussions/573).

Features we are planning for the longer term development of Langium:

* AST/semantic model serialization. I.e. transformation of an in-memory AST model back to DSL source code.
* Support for developing type systems for Langium based languages.
* General improvements on a grammar level (better validations, parser improvements, etc.)

**Comment by trusktr** [comment]
> * Support for developing type systems for Langium based languages.

That, sounds interesting!

**Reply by msujew** [reply]
We have a more detailed roadmap for each quarter. The current one can be found here https://github.com/langium/langium/discussions/863 (it's also pinned in the discussions board).
